crates/core/src/analysis.rs

```diff
@@ -1,22 +1,21 @@
-use crate::pattern_compiler::parse_one;
 use anyhow::{anyhow, Result};
-use grit_util::AstNode;
-use grit_util::{traverse, Order};
+use grit_util::{traverse, Ast, AstNode, Order};
+use marzano_language::grit_parser::MarzanoGritParser;
 use marzano_util::{cursor_wrapper::CursorWrapper, node_with_source::NodeWithSource};
 use std::collections::BTreeMap;
 use std::ffi::OsStr;
-use tree_sitter::Parser;
+use std::path::Path;
 
 /// Walks the call tree and returns true if the predicate is true for any node.
 /// This is potentially error-prone, so not entirely recommended
 fn walk_call_tree(
     node: &NodeWithSource,
     libs: &BTreeMap<String, String>,
-    grit_parser: &mut Parser,
+    grit_parser: &mut MarzanoGritParser,
```

Comment:
Introducing the `MarzanoGritParser` wasn't strictly necessary, because in the end it doesn't use the `Parser` trait either. But it does return the same new `Tree` data structure, so it helps make things a bit more consistent.